app/models/api/order_detail.rb

```diff
@@ -0,0 +1,35 @@
+class Api::OrderDetail
+  def initialize(order_detail)
+    @order_detail = order_detail
+  end
+
+  def to_h
+    { account: account_to_hash, ordered_for: ordered_for_to_hash }
+  end
+
+  private
+
+  def account
+    @account = @order_detail.try(:account) unless defined?(@account)
+    @account
+  end
+
+  def account_to_hash
+    if account.present?
+      { id: account.id, owner: user_to_hash(account.owner.user) }
+    end
+  end
+
+  def ordered_for_to_hash
+    user_to_hash(@order_detail.order.user)
+  end
+
+  def user_to_hash(user)
```

Comment:
Is it possible for an `Account` to have no owner, or an `Order` to have no user? I couldn't find either in the dev database.